### ISSUE TITLE
Add ability to pass arbitrary request params to API endpoint

### ISSUE THREAD

**Issue by DivineOmega** [langleyfoxall/react-dynamic-data-table#14]
We should allow devs to pass arbitrary request params to API endpoint when using the `AjaxDynamicDataTable` component. This could be used for to allow users to search/filter the results displayed.

Possible example syntax:

```jsx
<AjaxDynamicDataTable apiUrl={'http://example.test/web-api/users'} 
    params={[
        { searchQuery: 'test' },
        { createdAfter: '2018-10-01' },
    ]} />
```

**Comment by AlexCatch** [comment]
Don't forget headers as well

**Comment by DivineOmega** [comment]
Good suggestions. Perhaps there could be an `axios` parameter that lets you pass through your custom axios instance.

**Comment by DivineOmega** [comment]
@AlexCatch I've split up the `axios` prop idea into a seperate issue (#19).

**Comment by dextermb** [comment]
If we use the following structure for params, then it can just be passed into the axios options:
```jsx
{
   searchQuery: 'test',
   createdAfter: '2018-10-01',
}
```

**Comment by DivineOmega** [comment]
Fixed as part of #47.
